Multisig/UI/Assets/AssetsViewModel.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  AssetsViewModel.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 13.05.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+struct TokenBalance: Identifiable {
+    var id: String {
+        return address
+    }
+    var imageURL: URL {
+        return URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(address).png")!
+    }
+    let address: String
+    let balance: String
+    let balanceUsd: String
+}
+
+class AssetsViewModel: ObservableObject {
+    @Published var balances = [TokenBalance]()
+    @Published var isLoading: Bool = true
+    @Published var errorMessage: String? = nil
+
+    private var subscribers = Set<AnyCancellable>()
+
+    init() {
+        let request = Safe.fetchRequest().selected()
+        let selectedSafe = try! CoreDataStack.shared.viewContext.fetch(request).first!
+        isLoading = true
+        Just(selectedSafe.address!)
+            .setFailureType(to: Error.self)
+            .flatMap { address in
+                Future<[TokenBalance], Error> { promise in
+                    DispatchQueue.global().async {
+                        do {
+                            let balancesResponse = try App.shared.safeTransactionService.safeBalances(at: address)
+                            let tokenBalances = balancesResponse.map {
+                                TokenBalance(address: $0.tokenAddress ?? "0x0",
+                                             balance: $0.balance,
+                                             balanceUsd: $0.balanceUsd)
```

Comment:
What do you think about extracting these two methods into a business-logic method on the Safe, because this will be more complicated in the nearest future?

Reply:
I will make final adjustments with the final PR. Maybe yes, maybe no, let's see.